--- a/src/opcodescript/mod.rs
+++ b/src/opcodescript/mod.rs
@@ -80,12 +80,12 @@
 					// simple check for end.
 					if opcode.opcode() == 0x05
 						&& ([0x00, 0x05].contains(&data[address + 1])
-							|| 0x02
+							|| (!quirks.contains(Quirks::CCFC) && 0x02
 								== opcodes
 									.last()
 									.map(|it: &Opcode| it.opcode())
-									.unwrap_or_default()
-							|| (!quirks.contains(Quirks::LibraryParty)
+									.unwrap_or_default())
+							|| (!quirks.intersects(Quirks::LibraryParty | Quirks::CCFC)
 								&& (data.len() - address) < 0x30))
 					{
 						at_end = true;
